src/bvals/cc/bvals_cc_in_one.cpp

```diff
@@ -0,0 +1,553 @@
+      else
+        ej -= pmb->block_size.nx2 / 2 - pmb->cnghost;
+    } else {
+      if (nb.ni.fi2 == 1)
+        sj += pmb->block_size.nx2 / 2 - pmb->cnghost;
+      else
+        ej -= pmb->block_size.nx2 / 2 - pmb->cnghost;
+    }
+  }
+  if (nb.ni.ox3 == 0 && pmb->block_size.nx3 > 1) {
+    if (nb.ni.ox1 != 0 && nb.ni.ox2 != 0) {
+      if (nb.ni.fi1 == 1)
+        sk += pmb->block_size.nx3 / 2 - pmb->cnghost;
+      else
+        ek -= pmb->block_size.nx3 / 2 - pmb->cnghost;
+    } else {
+      if (nb.ni.fi2 == 1)
+        sk += pmb->block_size.nx3 / 2 - pmb->cnghost;
+      else
+        ek -= pmb->block_size.nx3 / 2 - pmb->cnghost;
+    }
+  }
+}
+
+// send boundary buffers with MeshBlockPack support
+// TODO(pgrete) should probaly be moved to the bvals or interface folders
+TaskStatus SendBoundaryBuffers(std::shared_ptr<MeshData<Real>> &md) {
+  Kokkos::Profiling::pushRegion("Task_SendBoundaryBuffers_MeshData");
+
+  auto var_pack = md->PackVariables(std::vector<MetadataFlag>({Metadata::FillGhost}));
+  const auto Nv = var_pack.GetDim(4);
+
+  Kokkos::Profiling::pushRegion("Reset boundaries");
+
+  int buffers_used = 0;
+  // reset buffer and count required buffers
+  for (int block = 0; block < md->NumBlocks(); block++) {
+    auto &rc = md->GetBlockData(block);
+    auto pmb = rc->GetBlockPointer();
+
+    for (auto &v : rc->GetCellVariableVector()) {
+      if (v->IsSet(parthenon::Metadata::FillGhost)) {
+        v->resetBoundary();
+      }
+    }
+
+    for (int n = 0; n < pmb->pbval->nneighbor; n++) {
+      parthenon::NeighborBlock &nb = pmb->pbval->neighbor[n];
+      // TODO(?) currently this only works for a single "Variable" per container.
+      // Need to update the buffer sizes so that it matches the packed Variables.
+      assert(rc->GetCellVariableVector().size() == 1);
+      auto *bd_var_ = rc->GetCellVariableVector()[0]->vbvar->GetPBdVar();
+      if (bd_var_->sflag[nb.bufid] == parthenon::BoundaryStatus::completed) continue;
+      buffers_used += 1;
+    }
+  }
+  Kokkos::Profiling::popRegion(); // Reset boundaries
+
+  auto boundary_info = md->GetSendBuffers();
+
+  if (!boundary_info.is_allocated() || true) {
+    Kokkos::Profiling::pushRegion("Create bndinfo array");
+
+    boundary_info = BufferCache_t("send_boundary_info", buffers_used);
+    auto boundary_info_h = Kokkos::create_mirror_view(boundary_info);
+
+    // now fill the buffer information
+    int b = 0; // buffer index
+    for (auto block = 0; block < md->NumBlocks(); block++) {
+      auto &rc = md->GetBlockData(block);
+      auto pmb = rc->GetBlockPointer();
+
+      int mylevel = pmb->loc.level;
+      for (int n = 0; n < pmb->pbval->nneighbor; n++) {
+        parthenon::NeighborBlock &nb = pmb->pbval->neighbor[n];
+        // TODO(?) currently this only works for a single "Variable" per container.
+        // Need to update the buffer sizes so that it matches the packed Variables.
+        assert(rc->GetCellVariableVector().size() == 1);
```

Comment:
maybe replace these asserts with `PARTHENON_REQUIRE` so we can hunt them down if we hit them at runtime.

Reply:
I plan to have this functionality implemented before we merge into develop, so this will go away eventually.